### PR TITLE
Docking system can no longer forcibly open welded doors.

### LINKED ISSUE #10714: Open welded airlocks
image

Welded external shuttle airlocks will still be forced open upon shuttle arrival, creating a funny graphical error.

Ideally the shuttle should check to see if the airlock is welded shut, and if it is welded the airlock should not automatically open until the welding job is undone.

### PR DESCRIPTION
**Pull request by Mervill**
## About the PR 

Fixes #10714

Docking system can no longer forcibly open welded doors.

**Changelog**

:cl:
- fix: Docking ports on shuttles no longer force welded doors open. Have fun welding the shuttle doors shut.

### CODE REVIEW

File: Content.Server/Shuttles/Systems/DockingSystem.cs [hunk 1]
```diff
@@ -363,13 +363,13 @@ public void Dock(DockingComponent dockA, DockingComponent dockB)
             if (TryComp(dockA.Owner, out DoorComponent? doorA))
             {
                 doorA.ChangeAirtight = false;
-                _doorSystem.StartOpening(doorA.Owner, doorA);
```
Comment (mirrorcult) [inline Content.Server/Shuttles/Systems/DockingSystem.cs:366]: im pretty sure these were `StartOpening` for a reason. furthermore, the code above bolts the airlocks before it starts opening them, so shouldnt this always fail if its checking now?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Shuttles/Systems/DockingSystem.cs (hunk 1 above)
Comment (Mervill) [reply Content.Server/Shuttles/Systems/DockingSystem.cs:366]: Solved the bolt problem by moving the bolt step too after the TryOpen step. Seems to work properly in testing.

File: Content.Server/Shuttles/Systems/DockingSystem.cs [hunk 2]
```diff
@@ -350,26 +350,26 @@ public void Dock(DockingComponent dockA, DockingComponent dockB)
...
+                doorB.ChangeAirtight = false;
+                _doorSystem.TryOpen(doorB.Owner, doorB);
             }
 
-            if (TryComp(dockA.Owner, out DoorComponent? doorA))
+            if (TryComp<AirlockComponent>(dockA.Owner, out var airlockA))
             {
-                doorA.ChangeAirtight = false;
-                _doorSystem.StartOpening(doorA.Owner, doorA);
+                airlockA.SetBoltsWithAudio(true);
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/DockingSystem.cs:367]: Shouldn't this check if the door successfully opened?

Also if we need ChangeAirtight then it should be handled upon undocking too.

**Review by metalgearsloth — DISMISSED** [review]

**Comment by Mervill** [comment]
ok I've done some more thorough testing and I think it's good to go now.

**Pull request merged**